mne/channels/channels.py

```diff
@@ -375,8 +376,10 @@ def set_channel_types(self, mapping):
                                  "fix the measurement info of your data."
                                  % (ch_name, unit_old))
             if unit_old != _human2unit[ch_type]:
-                warn("The unit for channel %s has changed from %s to %s."
-                     % (ch_name, _unit2human[unit_old], _unit2human[unit_new]))
+                this_change = (_unit2human[unit_old], _unit2human[unit_new])
+                if this_change not in unit_changes:
+                    unit_changes[this_change] = list()
+                unit_changes[this_change].append(ch_name)
```

Comment:
You think that's better? I find it a bit convoluted to set the dict content to a freshly constructed list each time, when what you're doing is appending.